src/test/regress/sql/multi_partitioning.sql

```diff
@@ -1162,6 +1162,45 @@ DROP TABLE partitioning_test, partitioning_test_2008, partitioning_test_2009,
            partitioning_test_2010, partitioning_test_2011,
            reference_table, reference_table_2;
 
+-- test cases for #3970
+
+--1. create a partitioned table
+CREATE TABLE part_table (
+    work_ymdt timestamp without time zone NOT NULL,
+    seq bigint NOT NULL,
+    my_seq bigint NOT NULL,
+    work_memo character varying(150),
+    CONSTRAINT work_memo_check CHECK ((octet_length((work_memo)::text) <= 150))
+)
+PARTITION BY RANGE (work_ymdt);
+
+--2. perform create_distributed_table
+SELECT create_distributed_table('part_table', 'seq');
+
+--3. add a partition
+CREATE TABLE part_table_p202008 PARTITION OF part_table FOR VALUES FROM ('2020-08-01 00:00:00') TO ('2020-09-01 00:00:00');
+
+--4. add a check constraint
+alter table part_table add CONSTRAINT my_seq CHECK (my_seq > 0);
```

Comment:
nit, uppercase SQL keywords
```suggestion
ALTER TABLE part_table ADD CONSTRAINT my_seq CHECK (my_seq > 0);
```